source/server/http/admin.cc

```diff
@@ -770,6 +769,14 @@ Http::Code AdminImpl::runCallback(absl::string_view path_and_query,
 
   for (const UrlHandler& handler : handlers_) {
     if (path_and_query.compare(0, query_index, handler.prefix_) == 0) {
+      if (handler.mutates_server_state_) {
+        const absl::string_view method =
+            admin_filter.requestHeaders().Method()->value().getStringView();
```

Comment:
Sorry I just noticed this. Is there any reason to pass the entire filter here vs. just passing the request headers? Seems a bit cleaner unless it's necessary (we can avoid accessor, etc.).

Reply:
Good question. What was on my mind here was eliminating the need for the dynamic_cast in https://github.com/envoyproxy/envoy/pull/2736/files/b9a4301d5b885ec2c244315791768edd68fa8352 in #2736 .

It makes sense to me to have admin handlers get more request context in general.  And in the case of that PR,  what's needed is ```Http::StreamDecoderFilterCallbacks* callbacks_```.  I'm also fine reducing the exposure now to just request-headers, and that could be broadened to the entire AdminFilter& in the future if needed.  I admit, though, I don't really know whether those callbacks_ are truly needed to solve the hystrix issues.

It's a question of speculative generality vs. reducing the need for an antipated refactor in the future, and I'm fine either way.

Reply:
My preference is to keep it simple for now unless there is a proven need, but I don't feel that strongly about it.

Reply:
ok done.